### ISSUE TITLE
Can't use env variable in mongodb.plugin.

### ISSUE THREAD

**Issue by Bosmak** [influxdata/influxdb#9746]
When writing a telegraf.conf using an env variable to define the connection URI for the MongoDB plugin, the authentication doesn't work.

`[[inputs.mongodb]]
   servers = ["$VAR"]`

Setting the $VAR to MongoDB://user:passwd@mongodb. (On a k8 yaml)

When running telegraf the plugin will connect to the MongoDB server but no authentication, the response from telegraf, in this case, is the same as if the user doesn't have the permission to run the desired query.

"2018-04-19T19:16:40.001+0000 I ACCESS   [conn7] Unauthorized: not authorized on admin to execute command { serverStatus: 1, recordStats: 0 }"
"2018-04-19T19:01:00Z E! Error in plugin [inputs.mongodb]: not authorized on admin to execute command { serverStatus: 1, recordStats: 0 }"

But when setting the same value as the var to the telegraf.conf directly it works just fine...

`[[inputs.mongodb]]
   servers = ["mongodb://user:passwd@mongodb"]`

This workaround has been found in the community forum. 
https://community.influxdata.com/t/telegraf-e-error-in-plugin-inputs-mongodb-not-authorized-on-admin-to-execute-command/4312

TL;DR If setting the MongoDB URI as an env variable authentication doesn't work.

**Comment by mark-rushakoff** [comment]
@Bosmak please file this issue on telegraf: https://github.com/influxdata/telegraf/issues/new
